src/query/expression/src/kernels/take.rs

```diff
@@ -144,6 +172,112 @@ impl Column {
         }
     }
 
+    pub fn take_by_compressd_indices(&self, indices: &[(u32, u32)], row_num: usize) -> Self {
+        let length = indices.len();
```

Comment:
add debug_assert! 

row_num must be the sum of indices.1

Reply:
done.